spec/models/course/assessment/assessment_ability_spec.rb

```diff
@@ -35,6 +44,8 @@
       it { is_expected.to be_able_to(:create, attempting_submission) }
       it { is_expected.to be_able_to(:update, attempting_submission) }
       it { is_expected.to be_able_to(:read, submitted_submission) }
+      it { is_expected.not_to be_able_to(:update, coursemate_attempting_submission) }
```

Comment:
There isn't a `show` action for `Course::Assessment::Submission::SubmissionsController`, so `:read, coursemate_attempting_submission` will not be triggered by that. The `edit` action does this check:
```ruby
  def authorize_submission!
    if @submission.attempting?
      authorize!(:update, @submission)
    else
      authorize!(:read, @submission)
    end
  end
```
These tests should be good enough?